irods/session.py

```diff
@@ -18,25 +20,32 @@
 from irods.exception import NetworkException
 from irods.password_obfuscation import decode
 from irods import NATIVE_AUTH_SCHEME, PAM_AUTH_SCHEME
-import threading
-import weakref
 from . import DEFAULT_CONNECTION_TIMEOUT
 
+_fds = None
+_fds_lock = threading.Lock()
```

Comment:
Are the `_fds` per-session or global across sessions? Will this have any adverse effects on existing code (i.e. deadlocks)?

If nothing comes to mind, please resolve this. Just making sure we've thought through possibilities here...

Reply:
The _fds are global but are each tied to an `iRODSConnection`.  A session may contain several of these.
No, currently I can't conceive of a situation where this (+ related code) would result in deadlock.